dqc_us_rules/util/facts.py

```diff
@@ -175,15 +175,36 @@ def LegalEntityAxis_facts_by_member(facts):
     results = defaultdict(list)
     for fact in facts:
         legalDim = LEGALENTITYAXIS_DEFAULT
-        dims = [dim for dim in fact.context.segDimValues.values() if dim.isExplicit and dim.member is not None]
-        for dim in dims:
-            if dim.dimension.qname.localName == 'LegalEntityAxis':
-                legalDim = dim.member.qname.localName
-                break
-        results[legalDim].append(fact)
+        if fact_components_valid(fact):
+            dims = [dim for dim in fact.context.segDimValues.values() if dim.isExplicit and dim.member is not None]
+            for dim in dims:
+                if dim.dimension.qname.localName == 'LegalEntityAxis':
+                    legalDim = dim.member.qname.localName
+                    break
+            results[legalDim].append(fact)
     return results
 
 
+def fact_components_valid(fact):
+    """
+    Return true if all of the components in a fact are none
```

Comment:
Think I've pointed this out before, but this docstring and the ```:return:``` string are misleading as you are returning ```True``` only if the fact components are NOT none.